apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -3794,6 +3805,42 @@ query_balances_(ConnPid, Accounts, <<"json-rpc">>) ->
             ConnPid, #{ <<"method">> => <<"channels.get.balances">>
                       , <<"params">> => #{<<"accounts">> => Accounts} })}.
 
+sc_ws_get_state(ConnPid, Config) ->
+    {ok, Res} = query_state(ConnPid, Config),
+    #{ <<"trees">> := EncodedTrees
+     , <<"calls">> := EncodedCalls
+     , <<"signed_tx">> := EncodedSignedTx
+     , <<"half_signed_tx">> := EncodedHalfSignedTx
+     } = Res,
+    {ok, STrees} = aehttp_api_encoder:safe_decode(state_trees, EncodedTrees),
+    Trees = aec_trees:deserialize_from_binary_without_backend(STrees),
+    {ok, SCalls} = aehttp_api_encoder:safe_decode(call_state_tree, EncodedCalls),
+    Calls = aect_call_state_tree:from_binary_without_backend(SCalls),
+    {ok, #{ trees => Trees
+          , calls => Calls
+          , signed_tx => decode_signed_tx(EncodedSignedTx)
+          , half_signed_tx => decode_signed_tx(EncodedHalfSignedTx)}}.
+
+decode_signed_tx(<<>>) ->
+    no_tx;
+decode_signed_tx(EncodedSignedTx) ->
+    {ok, SSignedTx} = aehttp_api_encoder:safe_decode(transaction, EncodedSignedTx),
+    aetx_sign:deserialize_from_binary(SSignedTx).
+
+query_state(ConnPid, Config) ->
+    query_state_(ConnPid, sc_ws_protocol(Config)).
+
+query_state_(ConnPid, <<"legacy">>) ->
+    %% get.offchain_state doesn't require any parameter, but not suplying one 
+    %% will remove "payload" key from request and cause error in legacy API
```

Comment:
Is this tracked in PT? It sounds like a bug.

Reply:
This is a bug in testing framework, it only affects "legacy" protocol. As we are removing the "legacy" protocol we probably don't need to fix this bug.